### PR TITLE
feat: add btc consolidation logic and new ui components to psbt and batch psbt screens

### PR DESCRIPTION
**Pull request by fedeerbes**
# 🔘 PR Type

- Enhancement

# 📜 Background

This PR adds the btc consolidation logic and new ui components to psbt and batch psbt screens

Issue Link: #[ENG-3393](https://linear.app/xverseapp/issue/ENG-3393/[ext-%F0%9F%8C%8D]-new-sign-psbt-screen-design-consistency-and-rare-sat)

# 🔄 Changes
- Add logic for psbt signing from btc consolidation logic in psbt screen
- Add new ui components to use new logic psbt screen
- Add logic and ui in batch psbt screen

Impact:
- psbt flow
- batch psbt flow

# 🖼 Screenshot / 📹 Video

Videos can be found in linear ticket

# ✅ Review checklist
Please ensure the following are true before merging:

- [ ] Code Style is consistent with the project guidelines.
- [ ] Code is readable and well-commented.
- [ ] No unnecessary or debugging code has been added.
- [ ] Security considerations have been taken into account.
- [ ] The change has been manually tested and works as expected.
- [ ] Breaking changes and their impacts have been considered and documented.
- [ ] Code does not introduce new technical debt or issues.

### CODE REVIEW

**Comment by dhriaznov** [comment]
I tested the batch psbt signing flow, and it looks good! The only thing is that inscription images take some time to load:

https://github.com/secretkeylabs/xverse-web-extension/assets/36603049/bd81ca28-65a5-49fa-b3b2-754c5b390eff

File: src/app/components/confirmBtcTransaction/utils.ts [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+      if (item.fromAddress === btcAddress) {
+        return itemsFromPayment.push(item);
+      }
+      if (item.fromAddress === ordinalsAddress) {
+        itemsFromOrdinal.push(item);
+      }
+    });
+
+    if (itemsFromOrdinal.length > 0) {
+      return outputsFromOrdinal.push(output);
```
Comment (m-aboelenein) [inline src/app/components/confirmBtcTransaction/utils.ts:73]: unnecessary return? 🤔

File: src/app/screens/signBatchPsbtRequest/index.tsx [hunk 2]
```diff
@@ -153,13 +162,21 @@ function SignBatchPsbtRequest() {
...
+    [txnContext],
   );
 
-  const parsedPsbts = useMemo(
-    () => payload.psbts.map(handlePsbtParsing),
-    [handlePsbtParsing, payload.psbts],
-  );
+  useEffect(() => {
+    (async () => {
+      const parsedPsbtsResult = await Promise.all(payload.psbts.map(handlePsbtParsing));
```
Comment (m-aboelenein) [inline src/app/screens/signBatchPsbtRequest/index.tsx:170]: can we make sure that we process the PSBTs in the same order we receive them here? the final response must match the requested order

File: src/app/components/confirmBtcTransaction/itemRow/amount.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          <NumericFormat
+            value={satsToBtc(new BigNumber(amount)).toFixed()}
+            displayType="text"
+            thousandSeparator
+            suffix=" BTC"
+            renderText={(value: string) => <StyledP typography="body_medium_m">{value}</StyledP>}
+          />
+          <StyledP typography="body_medium_s" color="white_400">
+            {getFiatAmountString(
+              getBtcFiatEquivalent(new BigNumber(amount), BigNumber(btcFiatRate)),
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/itemRow/amount.tsx:78]: agree, will take care of it

File: src/app/components/confirmBtcTransaction/itemRow/amountWithInscriptionSatribute.tsx [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+            // eslint-disable-next-line react/no-array-index-key
+            <div key={index}>
+              <Range>
+                <Inscription
+                  inscription={item}
+                  hideTypeSizeInfo
+                  onShowInscription={onShowInscription}
+                />
+              </Range>
+              {(satributesArray.length || inscriptions.length > index + 1) && (
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/itemRow/amountWithInscriptionSatribute.tsx:111]: yes, we show the divider if in between rows and if there are satributes items below

File: src/app/components/confirmBtcTransaction/itemRow/avatar.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import styled from 'styled-components';
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/itemRow/avatar.tsx:1]: sounds good!

File: src/app/components/divider/divider.tsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import styled from 'styled-components';
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/divider/divider.tsx:1]: sounds good!

File: src/app/components/confirmBtcTransaction/itemRow/bundle.tsx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+}));
+
+const AvatarContainer = styled.div`
+  margin-right: ${(props) => props.theme.space.xs};
+`;
+
+const RowsContainer = styled.div`
+  padding-left: ${(props) => props.theme.space.m};
+`;
+const Divider = styled.div`
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/itemRow/bundle.tsx:37]: this divider is a different one. is the one who links the items in a bundle 
Screenshot 2023-12-28 at 09 01 07

File: src/app/components/confirmBtcTransaction/itemRow/bundle.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
+import Link from '@assets/img/rareSats/link.svg';
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/itemRow/bundle.tsx:1]: make sense. will do that

File: src/app/components/confirmBtcTransaction/itemRow/inscriptionSatributeRow.tsx [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+    };
+  });
+
+  const amountOfExoticsOrInscribedSats = Object.values(satRanges).reduce(
+    (acc, range) => acc + range.totalSats,
+    0,
+  );
+
+  if (amountOfExoticsOrInscribedSats < amount) {
+    satRanges[-1] = {
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/itemRow/inscriptionSatributeRow.tsx:67]: we're adding a range for the common/unknown sats, cause they need to be displayed 
Screenshot 2023-12-28 at 09 10 44

File: src/app/components/confirmBtcTransaction/itemRow/inscriptionSatributeRow.tsx [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+    if (bHasInscriptions && bHasRareSatributes) return 1;
+
+    // sats inscribed and not rare above sats not inscribed and not rare
+    if (aHasRareSatributes) return -1;
+    if (bHasRareSatributes) return 1;
+
+    // equal ranges
+    return 0;
+  });
+  const totalExoticSats = satRangesArray.reduce(
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/itemRow/inscriptionSatributeRow.tsx:100]: good suggestion. will do that. thanks!

File: src/app/components/confirmBtcTransaction/index.tsx [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+        {!isBroadcast && <BroadcastCallout bodyText={t('PSBT_NO_BROADCAST_DISCLAIMER')} />}
+        <TransactionSummary
+          inputs={inputs}
+          outputs={outputs}
+          feeOutput={feeOutput}
+          isPartialTransaction={isPartialTransaction}
+          getFeeForFeeRate={getFeeForFeeRate}
+          onFeeRateSet={onFeeRateSet}
+          isSubmitting={isSubmitting}
+        />
```
[the remark this answers is not in the document]
Comment (fedeerbes) [reply src/app/components/confirmBtcTransaction/index.tsx:172]: the fees are in this prop `feeOutput={feeOutput}`. or you are referring to something else?

**Review by m-aboelenein — DISMISSED** [review]

**Review by dhriaznov — DISMISSED** [review]

**Review by m-aboelenein — APPROVED** [review]

**Review by DuskaT021 — APPROVED** [review]

**Pull request merged**